core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -22,12 +22,13 @@
 static void
 ipv4_addr_to_wasi_addr(uint32_t addr_num, uint16_t port, __wasi_addr_t *out)
 {
+    addr_num = htonl(addr_num);
```

Comment:
in L42 where calls `ipv4_addr_to_wasi_addr`, we pass `sin_addr.s_addr` as `addr_num`, and I think it is in "network byte order".

Reply:
good spot; `htonl` and `ntohl` are reverse operations so the code would still work well on both little and big endian, but I agree, better to not mix them. Updated the code